code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -114,6 +114,39 @@
 	if (istype(target, /obj/item) && target.loc == src) //no batong for radio birds
 		target.emp_act()
 
+/mob/living/critter/flock/weapon_attack(atom/target, obj/item/W, reach, params)
+	if (HAS_ATOM_PROPERTY(target, PROP_ATOM_FLOCK_THING) && target.density)
+		if (!reach && istype(W, /obj/item/gun))
+			boutput(src, "<span class='alert'>The grip tool refuses to harm this, jamming briefly.</span>")
+			return
+		return ..()
+	if (istype(W, /obj/item/gun))
+		if (src.a_intent == INTENT_GRAB && reach)
+			boutput(src, "<span class='alert'>You can't gun grab someone!</span>")
+			return
+		if (!reach || (reach && src.a_intent != INTENT_HELP))
+			if (prob(75))
+				boutput(src, "<span class='alert'>The grip tool can't quite reach the trigger!</span>")
+				playsound(src, 'sound/effects/brrp.ogg', 20, FALSE)
+				return
+			return ..()
+	else if (!reach)
+		return ..()
+	var/attack_force = W.force
+	var/attack_stam_hit = W.stamina_damage
+	W.force = attack_force / 5
+	W.stamina_damage = attack_stam_hit / 5
+	boutput(src, "<span class='alert'>The grip tool slips!</span>")
+	playsound(target, 'sound/effects/swoosh.ogg', 50, FALSE)
+	..()
+	if (!src.lastattacked)
+		src.lastattacked = target // prevents message and sound spam for when targetting unhittable stuff such as floors
+
+	SPAWN (1 MILLI SECOND)
```

Comment:
Good god no D: that is maximally cursed.

Can't you just call parent with a copy of `W` with its `force` changed?

Reply:
Better hope that item doesn't do anything funny in `New`

Reply:
Why use `SPAWN` anyway? Surely you want to change it back right after the attack proc.

Reply:
Tried that first but wasn't doing correct damage

Reply:
Might be the SPAWN(0) in /mob/attackby()